test/integration/fuzz/lockup-tranched/streamedAmountOf.t.sol

```diff
@@ -0,0 +1,148 @@
+    {
+        LockupTranched_Integration_Fuzz_Test.setUp();
+        StreamedAmountOf_Integration_Shared_Test.setUp();
+
+        // Disable the protocol fee so that it doesn't interfere with the calculations.
+        changePrank({ msgSender: users.admin });
+        comptroller.setProtocolFee({ asset: dai, newProtocolFee: ZERO });
+        changePrank({ msgSender: users.sender });
+    }
+
+    modifier givenMultipleTranches() {
+        _;
+    }
+
+    modifier whenCurrentTimestampNot1st() {
+        _;
+    }
+
+    /// @dev Given enough fuzz runs, all of the following scenarios will be fuzzed:
+    ///
+    /// - End time in the past
+    /// - End time in the present
+    /// - End time in the future
+    /// - Multiple deposit amounts
+    /// - Status streaming
+    /// - Status settled
+    function testFuzz_StreamedAmountOf_Calculation(
+        LockupTranched.Tranche[] memory tranches,
+        uint40 timeJump
+    )
+        external
+        givenNotNull
+        givenStreamHasNotBeenCanceled
+        whenStartTimeInThePast
+        givenMultipleTranches
+        whenCurrentTimestampNot1st
+    {
+        vm.assume(tranches.length > 1);
+
+        // Fuzz the tranche timestamps.
+        fuzzTrancheTimestamps(tranches, defaults.START_TIME());
+
+        // Fuzz the tranche amounts.
+        (uint128 totalAmount,) = fuzzTranchedStreamAmounts({
+            upperBound: MAX_UINT128,
+            tranches: tranches,
+            protocolFee: ZERO,
+            brokerFee: ZERO
+        });
+
+        // Bound the time jump.
+        uint40 firstTrancheDuration = tranches[1].timestamp - tranches[0].timestamp;
+        uint40 totalDuration = tranches[tranches.length - 1].timestamp - defaults.START_TIME();
+        timeJump = boundUint40(timeJump, firstTrancheDuration, totalDuration + 100 seconds);
+
+        // Mint enough assets to the Sender.
+        deal({ token: address(dai), to: users.sender, give: totalAmount });
+
+        // Create the stream with the fuzzed tranches.
+        LockupTranched.CreateWithTimestamps memory params = defaults.createWithTimestampsLT();
+        params.broker = Broker({ account: address(0), fee: ZERO });
+        params.tranches = tranches;
+        params.totalAmount = totalAmount;
+        uint256 streamId = lockupTranched.createWithTimestamps(params);
+
+        // Simulate the passage of time.
+        uint40 currentTime = defaults.START_TIME() + timeJump;
+        vm.warp({ timestamp: currentTime });
+
+        // Run the test.
+        uint128 actualStreamedAmount = lockupTranched.streamedAmountOf(streamId);
+        uint128 expectedStreamedAmount = calculateStreamedAmountForTranched(currentTime, tranches, totalAmount);
+        assertEq(actualStreamedAmount, expectedStreamedAmount, "streamedAmount");
+    }
+
+    /// @dev The streamed amount must never go down over time.
+    function testFuzz_StreamedAmountOf_Monotonicity(
```

Comment:
yes, it can be included in the invariant tests, but we faced some problems with the timestamps, and it was pretty complicated

@PaulRBerg can you confirm that i remember correctly?